File: backend_splatournament_manager/src/models/team.ts
export interface Team {
  id: number;
  name: string;
  tag: string;
  description: string;
  createdAt: string;
}

export interface TournamentTeam {
  id: number;
  tournamentId: number;
  teamId: number;
  registeredAt: string;
}

export interface TeamMember {
  id: number;
  teamId: number;
  userId: number;
  role: 'owner' | 'member';
  joinedAt: string;
}

File: frontend_splatournament_manager/lib/models/team.dart
class Team {
  final int id;
  final String name;
  final String tag;
  final String description;
  final String createdAt;
  final int? memberCount;

  Team({
    required this.id,
    required this.name,
    required this.tag,
    required this.description,
    required this.createdAt,
    this.memberCount,
  });

  factory Team.fromJson(Map<String, dynamic> json) {
    return Team(
      id: json['id'] as int,
      name: json['name'] as String,
      tag: json['tag'] as String,
      description: (json['description'] as String?) ?? '',
      createdAt: (json['createdAt'] as String?) ?? '',
      memberCount: json['memberCount'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'tag': tag,
      'description': description,
      'createdAt': createdAt,
      if (memberCount != null) 'memberCount': memberCount,
    };
  }
}

